cli/bpmetadata/path.go

```diff
@@ -0,0 +1,85 @@
+package bpmetadata
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+)
+
+const (
+	regexEx  = ".*/(examples/.*)"
+	regexMod = ".*/(modules/.*)"
+)
+
+func isPathValid(path string) (bool, error) {
+	_, err := os.ReadFile(path)
+	if err != nil {
+		return false, fmt.Errorf("unable to read file at the provided path: %v", err)
+	}
+
+	return true, nil
+}
+
+func getExamples(configPath string) ([]BlueprintMiscContent, error) {
+	re := regexp.MustCompile(regexEx)
+	return getDirPaths(configPath, re)
+}
+
+func getModules(configPath string) ([]BlueprintMiscContent, error) {
+	re := regexp.MustCompile(regexMod)
+	return getDirPaths(configPath, re)
+}
+
+func getDirPaths(configPath string, re *regexp.Regexp) ([]BlueprintMiscContent, error) {
+	dirContent, err := os.ReadDir(configPath)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read dir contents: %v", err)
+	}
+
+	var paths []BlueprintMiscContent
+	for _, c := range dirContent {
+		if c.IsDir() {
+			currDirPath := path.Join(configPath, c.Name())
+			// ignore .terraform directories
+			if strings.HasPrefix(c.Name(), ".terraform") {
+				continue
```

Comment:
Prefix may not match in the case of nested dir. We could do a contains to make this more robust.

Reply:
It would since `Name()` returns the name of the dir rather than the path. Even if it's nested, the name would start with ".terraform"

Reply:
Ah yes, I missed that. Thanks for clarifying.